subsys/caf/modules/sensor_sampler.c

```diff
@@ -223,7 +226,13 @@ static void sample_sensor(struct sensor_data *sd, const struct sensor_config *sc
 		LOG_ERR("Sensor sampling error (err %d)", err);
 		update_sensor_state(sc, sd, SENSOR_STATE_ERROR);
 	} else {
-		send_sensor_event(sc->event_descr, curr, ARRAY_SIZE(curr));
+		if (atomic_get(&sd->event_cnt) <= sc->active_events_limit) {
```

Comment:
`<` ? Otherwise we may submit up to `sc->active_events_limit +1` events at a time

Reply:
You mean because we decrement while event is still in workq as a last subscriber?

Reply:
`<` ? Discussed offline

Reply:
No, because we let user to submit one additional event. Discussed offline.